### PR TITLE
Add json parser

### PR DESCRIPTION
**Pull request by caldwecr**
First build of the gem to include functioning parser and high level validator

### CODE REVIEW

**Comment by caldwecr** [comment]
Reverted the removal of the gemfile so that circle picks up the dependencies without me having to run gem install --dev

File: lib/msfl/parsers/json.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+require 'json'
+module MSFL
+  module Parsers
+    class JSON
+      def self.parse(json)
```
Comment (caldwecr) [inline lib/msfl/parsers/json.rb]: Add spec and docs

File: lib/msfl/parsers/json.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        obj = ::JSON.parse(json)
+        obj = arrays_to_sets obj
+        obj
+      end
+
+      def self.arrays_to_sets(obj)
```
Comment (caldwecr) [inline lib/msfl/parsers/json.rb]: Add spec and docs

File: lib/msfl/validators/semantic.rb [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+    class Semantic
+      # Load definitions
+      include Definitions::HashKey
+
+      attr_accessor :dataset, :errors, :current_field, :current_operator
+
+      BOOLEAN_OPERATORS = [:and, :or]
+      ENUMERATION_OPERATORS = [:in]
+
+      def initialize(attributes = nil, opts = {})
```
Comment (caldwecr) [inline lib/msfl/validators/semantic.rb]: specs and docs

File: lib/msfl.rb [hunk 4]
```diff
@@ -1,5 +1,30 @@
...
+
+module MSFL
+  class << self
+    # Allows the user to set configuration options
+    #  by yielding the configuration block
+    #
+    # @param opts [Hash] an optional hash of options, supported options are `reset: true`
+    # @param block [Block] an optional configuration block
+    # @return [Configuration] the current configuration object
+    def configure(opts = {}, &block)
```
Comment (BlakeLucchesi) [inline lib/msfl.rb]: nice.

File: msfl.gemspec [hunk 5]
```diff
@@ -8,10 +8,30 @@ Gem::Specification.new do |s|
   s.email       = 'courtland@mattermark.com'
   s.files       = [
       "lib/msfl.rb",
+
```
Comment (BlakeLucchesi) [inline msfl.gemspec]: typical gem spec just lists lib/** if i recall correctly

**Pull request merged**